metadata-ingestion/src/datahub/cli/migration_utils.py

```diff
@@ -20,44 +21,68 @@
 
 log = logging.getLogger(__name__)
 
+all_aspects = [
+    "schemaMetadata",
+    "datasetProperties",
+    "viewProperties",
+    "subTypes",
+    "editableDatasetProperties",
+    "ownership",
+    "datasetDeprecation",
+    "institutionalMemory",
+    "editableSchemaMetadata",
+    "globalTags",
+    "glossaryTerms",
+    "upstreamLineage",
+    "datasetUpstreamLineage",
+    "status",
+    "containerProperties",
+    "dataPlatformInstance",
+    "containerKey",
+    "container",
+    "domains",
+    "containerProperties" "editableContainerProperties",
+]
+
 
 def get_aspect_name_from_relationship_type_and_entity(
     relationship_type: str, entity_type: str
 ) -> str:
-    if relationship_type == "Produces":
-        if entity_type.lower() == "datajob":
-            return "dataJobInputOutput"
-    if relationship_type == "Consumes":
-        if entity_type.lower() == "chart":
-            return "chartInfo"
-        if entity_type.lower() == "datajob":
-            return "dataJobInputOutput"
-        if entity_type.lower() == "dataProcess":
-            return "dataProcessInfo"
-    if relationship_type == "DownstreamOf":
-        if entity_type.lower() == "dataset":
-            return "upstreamLineage"
-    if relationship_type == "ForeignKeyToDataset":
-        if entity_type.lower() == "dataset":
-            return "schemaMetadata"
-    if relationship_type == "DerivedFrom":
-        if entity_type.lower() == "mlfeature":
-            return "mlFeatureProperties"
-        if entity_type.lower() == "mlprimarykey":
-            return "mlPrimaryKeyProperties"
+    aspect_map = {
+        "Produces": {"datajob": "dataJobInputOutput"},
+        "Consumes": {
+            "chart": "chartInfo",
+            "datajob": "dataJobInputOutput",
+            "dataProcess": "dataProcessInfo",
+        },
+        "DownstreamOf": {"dataset": "upstreamLineage"},
+        "ForeignKeyToDataset": {"dataset": "schemaMetadata"},
+        "DerivedFrom": {
+            "mlfeature": "mlFeatureProperties",
+            "mlprimarykey": "mlPrimaryKeyProperties",
+        },
+        "IsPartOf": {"container": "container", "dataset": "container"},
+    }
+
+    if (
+        relationship_type in aspect_map
+        and entity_type.lower() in aspect_map[relationship_type]
+    ):
+        return aspect_map[relationship_type][entity_type.lower()]
+
     raise Exception(
         f"Unable to map aspect name from relationship_type {relationship_type} and entity_type {entity_type}"
     )
 
 
-def modify_urn_list_for_aspect(
-    aspect_name: str,
-    aspect: DictWrapper,
-    relationship_type: str,
-    old_urn: str,
-    new_urn: str,
-) -> DictWrapper:
-    if aspect_name == "dataJobInputOutput":
+class UrnListModifier:
```

Comment:
just curious - why is this better?

Reply:
Flake8 is complaining about the high complexity of the method because of the too many if branches.